### DISCUSSION TITLE
get_next_page() function explain

### DISCUSSION THREAD

**Discussion by Drzhivago264** [kevinzg/facebook-scraper#813] [Q&A] [answered]
Sorry for my slow-witted. 
Can anyone explain to me the process of page iteration?
I guess it must be more complicated than automatically scrolling the pages as a selenium bot.  
If I am not incorrect, it searches in the current responded HTML to check whether something is matched or not. But I cannot find in the responded HTML of a post any information about the next post.
It is so great if anyone can give me a hint of where to look at.
Best regards,
Z

**Comment by Drzhivago264** [comment, marked as answer]
*Update: I have found this from the HTML of a page, for example, Cocacola:
href:"/page_content_list_view/more/?page_id=119565995152&start_cursor=%7B%22timeline_cursor%22%3A%22AQHRPh9D3rGzKZXld_2Mb4kl4AERRxTjL7eHIJsy08GoC1FWKAUjDaSri26B_LTJuUDAAtFlrhhlbZyqcos5l-isHGIydQCwSOl9zBYi03wFAsD56B3xwnO8rif0WXVcxjVL%22%2C%22timeline_section_cursor%22%3Anull%2C%22has_next_page%22%3Atrue%7D&num_to_fetch=4&surface_type=posts_tab"

Can you please tell me what the scrapper does with this? where is the data of the next page.
Thank you

**Reply by neon-ninja** [reply]
Hello. This scraper doesn't use selenium. Selenium needs a browser to run, which involves significant overhead, so is typically slower than just making web requests. The scraper requests HTML from Facebook servers and parses it. As you found, the scraper looks for URLs like that one, and makes a GET request to the URL. This kind of URL returns JSON containing HTML. The HTML can be parsed to extract post information.

**Reply by Drzhivago264** [reply]
Thank you I understand now.
A quick question:
How do you find such information in the JavaScript portion of Facebook HTML?
I found it in your code but is this a common practice on other social media.
I'm asking because I'm trying to build a small site. It looks like Facebook generates a unique hash for each of its pagination pages. 
Best regards,
Z

**Reply by neon-ninja** [reply]
With regex. Yes.
